proto/substrait/validator/simple_extensions.proto

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: Apache-2.0
+syntax = "proto3";
+
+// This proto file defines a machine-readable form of simple extension YAML
+// files.
+
+package substrait.validator;
+
+import "google/protobuf/empty.proto";
+import "substrait/validator/type_system.proto";
+
+option csharp_namespace = "Substrait.Validator.Protobuf";
+option java_multiple_files = true;
+option java_package = "io.substrait.validator.proto";
+
+// Root message returned by the validator as a result of converting a simple
+// extension module to machine-readable form.
+message SimpleExtension {
+  // Metadata about the simple extension module.
+  ExtensionDefinition module = 1;
+
+  // All extensions defined in the type class namespace of this extension. All
+  // names are case-insensitively unique.
+  repeated ExtensionDefinition type_classes = 2;
```

Comment:
How about making this (and the same for the other fields) `repeated TypeClass` where `message TypeClass { ExtensionMetaData metadata, ... }`, and `message ExtensionMetaData { string uri = 1; repeated string names = 2; ... }`? This would make consuming these messages more type safe: no need to match on the kind (it should match but there is no guarantee by the given structure).

Reply:
Refactored, but it kinda rippled all over the place thanks to there being different function extension types.